SnapWar/ViewControllers/SWAuthViewController.swift

```diff
@@ -28,21 +29,66 @@ class SWAuthViewController: UIViewController {
         formContainer.isLayoutMarginsRelativeArrangement = true
         formContainer.layoutMargins = UIEdgeInsets(top: 0, left: horizontalMargin, bottom: 0, right: horizontalMargin)
         
-        let keyboardTypes = [
-            UIKeyboardType.alphabet,
-            UIKeyboardType.numberPad,
-            UIKeyboardType.emailAddress
-        ]
-        for keyboardType in keyboardTypes {
-            let textField = SWTextField()
-            textField.keyboardType = keyboardType
-            textField.placeholder = "type here"
+        usernameField.placeholder = "username"
+        usernameField.delegate = self
+        usernameField.tag = 0
+        formContainer.addArrangedSubview(usernameField)
 
-            formContainer.addArrangedSubview(textField)
-        }
-        
-        let passwordField = SWPasswordTextField()
         passwordField.placeholder = "password"
+        passwordField.delegate = self
+        passwordField.tag = 1
         formContainer.addArrangedSubview(passwordField)
+        
+        loginButton.setTitle("Login", for: .normal)
+        loginButton.backgroundColor = SWColorType.primary
+        loginButton.addTarget(self, action: #selector(handleSubmit), for: .touchUpInside)
+        setLoginButtonState()
+        formContainer.addArrangedSubview(loginButton)
+    }
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
```

Comment:
i like to have delegate methods separated out from our code by using extensions, something like: https://github.com/SnapWars/SnapWar/blob/23ad07b5c120b0e1a4cbe4ed2205e1ad332c9365/SnapWarShareExtension/SWSharePageViewController.swift#L116

Reply:
the reason is to make code less cluttered, and make it easier to find these delegate methods by applying grouping on them, as later on you might implement more delegate protocols